Library/Homebrew/cmd/doctor.rb

```diff
@@ -428,6 +428,14 @@ def check_tmpdir_sticky_bit
   end
 end
 
+def check_supported_osx
+  if MacOS.version == "10.11" then <<-EOS.undent
+    Although Homebrew has added initial recognition of 10.11 it is unsupported.
+    You may encounter breakage or other failure and there is no guarantee
+    Homebrew can resolve those issues until El Capitan is stable.
+  EOS
+  end
+end
```

Comment:
Every and any time a `brew doctor` is run. I can move it to `brew.rb` instead though? We already have a too-old OS X version warning in there, so a warning about too-new wouldn't be out of place.